r1cs/gadgets/std/src/fields/nonnative/nonnative_field_mul_result_gadget.rs

```diff
@@ -76,13 +131,69 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField>
         Ok(res)
     }
 
-    /// Constraints for reducing the result of a multiplication mod p, to get an original representation.
+    /// Reducing a NonNativeMulResultGadget back to a non-native field gadget
+    /// in normal form. Assumes that 
+    /// ``
+    ///     2 * bits_per_limb + surfeit' <= CAPACITY - 2.
```

Comment:
If really, as described in group_and_check_equality, the carries don't affect length bounds, we should have <= CAPACITY - 1. Anyway now I think that carries **do** affect the bounds, so it is fine.